packages/frontend/src/components/common/SpaceActionModal/CreateSpaceModalContent.tsx

```diff
@@ -1,17 +1,110 @@
-import { FC } from 'react';
-import { SpaceModalBody } from '.';
+import { Radio } from '@blueprintjs/core';
+import { FC, useState } from 'react';
+import { CreateSpaceModalBody } from '.';
+import { useApp } from '../../../providers/AppProvider';
 import Input from '../../ReactHookForm/Input';
+import RadioGroup from '../../ReactHookForm/RadioGroup';
+import {
+    FlexWrapper,
+    PrimaryText,
+    UserCircle,
+    UserRole,
+    YouLabel,
+} from '../ShareSpaceModal/ShareSpaceModal.style';
+import {
+    AccessOption,
+    SpaceAccessOptions,
+    SpaceAccessType,
+} from '../ShareSpaceModal/ShareSpaceSelect';
+import { CreateSpaceAddUser } from './CreateSpaceAddUser';
+import { CreateSpaceSelectAccessType } from './CreateSpaceSelectAccessType';
+import { RadioDescription, ShareSpaceWrapper } from './SpaceActionModal.style';
 
-const CreateSpaceModalContent: FC<SpaceModalBody> = () => {
-    return (
-        <Input
-            label="Enter a memorable name for your space"
-            name="name"
-            placeholder="eg. KPIs"
-            defaultValue=""
-            rules={{ required: 'Name field is required' }}
-        />
+const CreateSpaceModalContent: FC<CreateSpaceModalBody> = ({
+    modalStep,
+    form,
+    setIsShared,
+}) => {
+    const {
+        user: { data: sessionUser },
+    } = useApp();
+    const [selectedAccess, setSelectedAccess] = useState<AccessOption>(
+        SpaceAccessOptions[0],
     );
+
+    switch (modalStep) {
+        case 0:
```

Comment:
can you change modalStep to be enum type instead of number? that will help with readability